File: wallet.php
<?php

class Wallet {
    private $solde;
    private $documents;

    public function __construct($solde_initial = 0) {
        $this->solde = $solde_initial;
        $this->documents = [
            'cin' => null,
            'carteBancaire' => null,
            'permis' => null
        ];
    }

    public function afficherSolde() {
        echo "Solde actuel: {$this->solde} euros\n";
    }

    public function deposer($montant) {
        $this->solde += $montant;
        echo "Dépôt de {$montant} euros effectué avec succès.\n";
        $this->afficherSolde();
    }

    public function retirer($montant) {
        if ($montant <= $this->solde) {
            $this->solde -= $montant;
            echo "Retrait de {$montant} euros effectué avec succès.\n";
            $this->afficherSolde();
        } else {
            echo "Solde insuffisant pour effectuer le retrait.\n";
        }
    }

    public function afficherDocuments() {
        echo "Documents actuels:\n";
        foreach ($this->documents as $document => $valeur) {
            echo "{$document}: " . ($valeur ? "Présent" : "Absent") . "\n";
        }
    }

    public function ajouterDocument($type, $document) {
        if (array_key_exists($type, $this->documents)) {
            $this->documents[$type] = $document;
            echo "{$type} ajouté avec succès.\n";
            $this->afficherDocuments();
        } else {
            echo "Type de document invalide.\n";
        }
    }

    public function retirerDocument($type) {
        if (array_key_exists($type, $this->documents)) {
            $this->documents[$type] = null;
            echo "{$type} retiré avec succès.\n";
            $this->afficherDocuments();
        } else {
            echo "Type de document invalide.\n";
        }
    }
}

function executerCommandes() {
    $wallet = new Wallet();

    while (true) {
        echo "Entrez une commande (solde, depot, retrait, documents, quitter): ";
        $commande = strtolower(trim(fgets(STDIN)));

        switch ($commande) {
            case "solde":
                $wallet->afficherSolde();
                break;
            case "depot":
                echo "Entrez le montant du dépôt : ";
                $montantDepot = floatval(trim(fgets(STDIN)));
                $wallet->deposer($montantDepot);
                break;
            case "retrait":
                echo "Entrez le montant du retrait : ";
                $montantRetrait = floatval(trim(fgets(STDIN)));
                $wallet->retirer($montantRetrait);
                break;
            case "documents":
                echo "Voulez-vous ajouter ou retirer un document ? ";
                $action = strtolower(trim(fgets(STDIN)));
                if ($action == "ajouter") {
                    echo "Entrez le type de document (CIN, carteBancaire, permis) : ";
                    $typeDocument = strtolower(trim(fgets(STDIN)));
                    echo "Entrez le numéro de {$typeDocument} : ";
                    $numeroDocument = trim(fgets(STDIN));
                    $wallet->ajouterDocument($typeDocument, $numeroDocument);
                } elseif ($action == "retirer") {
                    echo "Entrez le type de document à retirer (CIN, carteBancaire, permis) : ";
                    $typeDocument = strtolower(trim(fgets(STDIN)));
                    $wallet->retirerDocument($typeDocument);
                } else {
                    echo "Action invalide.\n";
                }
                break;
            case "quitter":
                echo "Fermeture du programme.\n";
                return;
            default:
                echo "Commande invalide. Veuillez réessayer.\n";
        }
    }
}

executerCommandes();

File: wallet.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class Wallet {
    private double balance;
    private Map<String, String> documents;

    public Wallet(double initialBalance) {
        this.balance = initialBalance;
        this.documents = new HashMap<>();
        this.documents.put("cin", null);
        this.documents.put("carteBancaire", null);
        this.documents.put("permis", null);
    }

    public void displayBalance() {
        System.out.println("Solde actuel: " + this.balance + " euros");
    }

    public void deposit(double amount) {
        this.balance += amount;
        System.out.println("Dépôt de " + amount + " euros effectué avec succès.");
        displayBalance();
    }

    public void withdraw(double amount) {
        if (amount <= this.balance) {
            this.balance -= amount;
            System.out.println("Retrait de " + amount + " euros effectué avec succès.");
            displayBalance();
        } else {
            System.out.println("Solde insuffisant pour effectuer le retrait.");
        }
    }

    public void displayDocuments() {
        System.out.println("Documents actuels:");
        for (Map.Entry<String, String> entry : documents.entrySet()) {
            String document = entry.getKey();
            String value = entry.getValue() != null ? "Présent" : "Absent";
            System.out.println(document + ": " + value);
        }
    }

    public void addDocument(String type, String document) {
        if (documents.containsKey(type)) {
            documents.put(type, document);
            System.out.println(type + " ajouté avec succès.");
            displayDocuments();
        } else {
            System.out.println("Type de document invalide.");
        }
    }

    public void removeDocument(String type) {
        if (documents.containsKey(type)) {
            documents.put(type, null);
            System.out.println(type + " retiré avec succès.");
            displayDocuments();
        } else {
            System.out.println("Type de document invalide.");
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        Wallet wallet = new Wallet(0);

        while (true) {
            System.out.print("Entrez une commande (solde, depot, retrait, documents, quitter): ");
            String command = scanner.nextLine().toLowerCase();

            switch (command) {
                case "solde":
                    wallet.displayBalance();
                    break;
                case "depot":
                    System.out.print("Entrez le montant du dépôt : ");
                    double depositAmount = scanner.nextDouble();
                    scanner.nextLine(); // Pour consommer le retour à la ligne
                    wallet.deposit(depositAmount);
                    break;
                case "retrait":
                    System.out.print("Entrez le montant du retrait : ");
                    double withdrawAmount = scanner.nextDouble();
                    scanner.nextLine(); // Pour consommer le retour à la ligne
                    wallet.withdraw(withdrawAmount);
                    break;
                case "documents":
                    System.out.print("Voulez-vous ajouter ou retirer un document ? ");
                    String action = scanner.nextLine().toLowerCase();
                    if (action.equals("ajouter")) {
                        System.out.print("Entrez le type de document (CIN, carteBancaire, permis) : ");
                        String documentType = scanner.nextLine().toLowerCase();
                        System.out.print("Entrez le numéro de " + documentType + " : ");
                        String documentNumber = scanner.nextLine();
                        wallet.addDocument(documentType, documentNumber);
                    } else if (action.equals("retirer")) {
                        System.out.print("Entrez le type de document à retirer (CIN, carteBancaire, permis) : ");
                        String documentType = scanner.nextLine().toLowerCase();
                        wallet.removeDocument(documentType);
                    } else {
                        System.out.println("Action invalide.");
                    }
                    break;
                case "quitter":
                    System.out.println("Fermeture du programme.");
                    scanner.close();
                    return;
                default:
                    System.out.println("Commande invalide. Veuillez réessayer.");
            }
        }
    }
}
